engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -459,44 +431,34 @@ private void markCurrentBeanAsProcessedForCurrentGroup(Object bean, Class<?> gro
 	public static class ValidationContextBuilder {
 		private final BeanMetaDataManager beanMetaDataManager;
 		private final ConstraintValidatorManager constraintValidatorManager;
-		private final MessageInterpolator messageInterpolator;
 		private final ConstraintValidatorFactory constraintValidatorFactory;
 		private final TraversableResolver traversableResolver;
-		private final ClockProvider clockProvider;
 		private final HibernateConstraintValidatorInitializationContext constraintValidatorInitializationContext;
-		private final boolean failFast;
+		private final ValidatorContextProperties validatorContextProperties;
 
 		private ValidationContextBuilder(
 				BeanMetaDataManager beanMetaDataManager,
 				ConstraintValidatorManager constraintValidatorManager,
-				MessageInterpolator messageInterpolator,
 				ConstraintValidatorFactory constraintValidatorFactory,
-				TraversableResolver traversableResolver,
-				ClockProvider clockProvider,
 				HibernateConstraintValidatorInitializationContext constraintValidatorInitializationContext,
-				boolean failFast) {
+				ValidatorContextProperties validatorContextProperties) {
 			this.beanMetaDataManager = beanMetaDataManager;
 			this.constraintValidatorManager = constraintValidatorManager;
-			this.messageInterpolator = messageInterpolator;
 			this.constraintValidatorFactory = constraintValidatorFactory;
-			this.traversableResolver = traversableResolver;
-			this.clockProvider = clockProvider;
+			this.traversableResolver = validatorContextProperties.getWrapedWithCachingForSingleValidation();
```

Comment:
I don't think it's a good idea to incorporate the `TraversableResolver` in it. It might not be `Validator` scoped so I don't think it's such a good idea to make it look like it.